### ISSUE TITLE
OneToManySubjectBuilder issue in TreeEntity

### ISSUE THREAD

**Issue by kurpav** [typeorm/typeorm#5357]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.22` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

When I try to save Product group with related products I'm getting the following:
```
TypeError: Cannot read property 'find' of undefined
    at /Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:119:90
    at Array.forEach (<anonymous>)
    at OneToManySubjectBuilder.buildForSubjectRelation (/Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:78:25)
    at /Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:42:22
    at Array.forEach (<anonymous>)
    at /Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:36:49
    at Array.forEach (<anonymous>)
    at OneToManySubjectBuilder.build (/Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:35:23)
    at EntityPersistExecutor.<anonymous> (/Users/kurpav/Work/foodhub/foodhub-backend/src/persistence/EntityPersistExecutor.ts:104:63)
    at step (/Users/kurpav/Work/foodhub/foodhub-backend/node_modules/tslib/tslib.js:136:27)
```

```ts
@Entity()
@Tree('nested-set')
export class ProductGroup extends BaseEntity {
  @Expose()
  @Column()
  name: string;

  @Expose()
  @Column()
  description: string;

  @OneToMany(
    type => Product,
    product => product.group,
    { cascade: true },
  )
  products: Product[];

  @TreeChildren()
  children: ProductGroup[];

  @TreeParent()
  parent: ProductGroup;
}
```

```ts
@Entity()
export class Product extends BaseEntity {
  @Expose()
  @Column()
  name: string;

  @Expose()
  @Column()
  description: string;

  @Expose()
  @Column()
  expiryDate: Date;

  @Expose()
  @Column()
  isAllergen: boolean;

  @Expose()
  @Column()
  nutritionalValueNotes: string;

  @Expose()
  @Column()
  supplierLeadTime: number;

  @ManyToOne(
    type => ProductGroup,
    group => group.products,
  )
  group: ProductGroup;
}
```

**Comment by benbierens** [comment]
I think I'm having the exact same problem.

version "0.2.22"

This error seems to occure only sometimes. We're running automated integration tests and this error comes up once every 3 to 5 runs on average.

`
(node:17363) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'find' of undefined
    at /Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:119:90
    at Array.forEach (<anonymous>)
    at OneToManySubjectBuilder.buildForSubjectRelation (/Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:78:25)
    at /Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:42:22
    at Array.forEach (<anonymous>)
    at /Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:36:49
    at Array.forEach (<anonymous>)
    at OneToManySubjectBuilder.build (/Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/subject-builder/OneToManySubjectBuilder.ts:35:23)
    at EntityPersistExecutor.<anonymous> (/Users/phnlnlv24135/data/Cardio/control-tower-backend/src/persistence/EntityPersistExecutor.ts:104:63)
    at step (/Users/phnlnlv24135/data/Cardio/control-tower-backend/node_modules/tslib/tslib.js:136:27)
(node:17363) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:17363) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

**Comment by AlexMesser** [comment]
This issue was probably fixed in #7156. Please open a new issue if there is still a problem.
